cloud-sql/sql-server/Startup.cs

```diff
@@ -44,55 +43,99 @@ public Startup(IConfiguration configuration)
         // http://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
-            string projectId = Google.Api.Gax.Platform.Instance().ProjectId;
-            if (!string.IsNullOrEmpty(projectId))
-            {
-                services.AddGoogleExceptionLogging(options =>
-                {
-                    options.ProjectId = projectId;
-                    options.ServiceName = "CloudSqlSample";
-                    options.Version = "Test";
-                });
-            }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddSingleton(sp => new SqlConnection().GetSqlServerConnectionString());
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
-            _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env,
+            ILoggerFactory loggerFactory)
         {
-            DbConnection connection;
-            string database = Configuration["CloudSQL:Database"];
-            connection = GetSqlServerConnection();
-            connection.Open();
-            using (var createTableCommand = connection.CreateCommand())
+            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+
+            if (env.IsDevelopment())
             {
-                // Create the 'votes' table if it does not already exist.
-                createTableCommand.CommandText = @"
-                 IF OBJECT_ID(N'dbo.votes', N'U') IS NULL
-                   BEGIN
-                     CREATE TABLE dbo.votes(
-                       vote_id INT NOT NULL IDENTITY(1, 1) PRIMARY KEY,
-                       time_cast datetime NOT NULL,
-                       candidate CHAR(6) NOT NULL)
-                   END";
-                createTableCommand.ExecuteNonQuery();
+                app.UseDeveloperExceptionPage();
             }
-            return connection;
+            else
+            {
+                // Configure error reporting service.
+                app.UseGoogleExceptionLogging();
+                app.UseExceptionHandler("/Home/Error");
+            }
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
         }
+    }
+
+    static class StartupExtensions
+    {
+        public static DbConnection OpenWithRetry(this DbConnection connection) =>

```

Comment:
And similar Polly config changes here, and InitializeDatabase related changes.

Reply:
Done.